sw/src/lib/registration.js

```diff
@@ -0,0 +1,132 @@
+export async function checkForUpdates(onUpdate) {
```

Comment:
This is an addition to the CRA registration to decouple registration from update handling -- the update logic might need some changes when the Fatal Error Boundary in the app adapter attempts to update the service worker